pyupgrade.py

```diff
@@ -57,26 +108,29 @@ def _convert_tup(tup):
     return j.join(_convert_tup(tup) for tup in parsed)
 
 
-def _ast_to_offset(node):
+def _ast_to_offset(node):  # type: (Union[ast.expr, ast.stmt]) -> Offset
     return Offset(node.lineno, node.col_offset)
 
 
-def ast_parse(contents_text):
+def ast_parse(contents_text):  # type: (Text) -> ast.Module
     # intentionally ignore warnings, we might be fixing warning-ridden syntax
     with warnings.catch_warnings():
-        warnings.simplefilter('ignore')
-        return ast.parse(contents_text.encode('UTF-8'))
+        warnings.simplefilter('ignore')  # type: ignore
+        ast_module = ast.parse(contents_text.encode('UTF-8'))
 
+    assert isinstance(ast_module, ast.Module)
+    return ast_module
 
-def inty(s):
+
+def inty(s):  # type: (Any) -> bool
```

Comment:
should be `# type: (str) -> bool`

Reply:
i tried to keep it generic

Reply:
since it gracefully handles non-strs. also note the caller could pass null originally.

Reply:
hmmm, how about `Optional[str] -> bool` then?

Reply:
I can change it but I'm not sure I agree. Shouldn't typing be as loose as possible based on the function's internals, instead of on how the callers use it? For example suppose we have this function which is always called with a list of ints

```
def mysum(xs):
    return sum(x for x in xs)
```

It could be typed as `(List[int]) -> int` and be correct in the scope of the existing program but isn't it more extensible to type it as `(Iterable[int]) -> int`? (assuming that only iterables of ints are ever passed and not custom objects with `__radd__` impls). 

Similarly here, `Any` is a valid input based on the function's implementation, even though current callers only pass `Optional[str]`. We don't type super generically everywhere because it can make the code overly generic & hard to read but I think it makes sense to do it in places like this.

Reply:
I had to do the None check in the caller so that `int` wouldn't complain. So maybe this can just be `str` after all.

Another option is to create a `maybe_int` function which returns the int if it can be done and some bogus int if it's not possible, and use that for comparisons. But that has other problems